pkg/provider/cloud/aks/provider.go

```diff
@@ -225,7 +237,7 @@ func ConvertStatus(provisioningState string, powerState armcontainerservice.Code
 	case provisioningState == string(resources.SucceededAKSState) && powerState == armcontainerservice.Code(resources.StoppedAKSState):
 		return apiv2.StoppedExternalClusterState
 	case provisioningState == string(resources.FailedAKSState):
```

Comment:
lets remove this condition for error and replace it with warning:
`case provisioningState == string(resources.FailedAKSState) && string(powerState) == string(resources.RunningAKSMDState)`